python/metatensor-learn/tests/sequential.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+
+import metatensor
+
+
+torch = pytest.importorskip("torch")
+
+from metatensor.learn import nn as nn  # noqa: E402
+
+from ._rotation_utils import WignerDReal  # noqa: E402
+
+
+@pytest.fixture
+def tensor():
+    tensor = metatensor.load(
+        "../metatensor-operations/tests/data/qm7-spherical-expansion.npz",
+        use_numpy=True,
+    ).to(arrays="torch")
+    tensor = metatensor.remove_gradients(tensor)
+    return tensor
+
+
+@pytest.fixture
+def wigner_d_real():
+    return WignerDReal(lmax=4, angles=(0.87641, 1.8729, 0.9187))
+
+
+def test_sequential_mlp(tensor):
+    """
+    Constructs a small multi-layer perceptron from standard metatensor nn modules
+    """
+    in_keys = tensor.keys
+    in_features = [len(tensor.block(key).properties) for key in in_keys]
+
+    model = nn.Sequential(
+        in_keys,
+        nn.LayerNorm(
+            in_keys=in_keys,
+            in_features=in_features,
+            dtype=torch.float64,
+        ),
+        nn.Linear(
+            in_keys=in_keys,
+            in_features=in_features,
+            out_features=4,
+            bias=True,
+            dtype=torch.float64,
+        ),
+        nn.Tanh(in_keys=in_keys),
+        nn.Linear(
+            in_keys=in_keys,
+            in_features=4,
+            out_features=1,
+            bias=True,
+            dtype=torch.float64,
+        ),
+    )
+
+    prediction = model(tensor)
+    assert metatensor.equal_metadata(
```

Comment:
I think this should also check that the values are the same, at least for one block.

Reply:
Check what values are the same sorry? I have no reference data here, should I write a regression test?

Reply:
I meant writing the same sequential for normal torch tensor & sending a single block through. But with random initialization, this might be a lot of complexity and not much advantages, so we can maybe do without